rustc: add wasm32v1-none target

Rust 1.84 includes a new target wasm32v1-none (https://doc.rust-lang.org/beta/rustc/platform-support/wasm32v1-none.html). This target could be used by some of the packages that are currently targeting `wasm32-unknown-unknown`, namely polkadot which I maintain (https://github.com/paritytech/polkadot-sdk/pull/7008). I added the target to rustc by adding `wasm32v1-none` [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/rust/rustc.nix#L164). Unfortunately this did not work since the rustc build system will by default try to build docs for all targets, including docs for std which will fail for `wasm32v1-none` since it's a no-std target. To workaround this I tried building rustc with `--disable-docs` which worked and the `wasm32v1-none` worked correctly to build polkadot. Not sure how to tune the rustc build to only build docs for std targets.

cc @alyssais

```
rustc> Documenting stage2 library {alloc, core, panic_abort, panic_unwind, proc_macro, profiler_builtins, std, sysroot, test, unwind} in HTML format (x86_64-unknown-linux-gnu -> wasm32v1-none)
rustc>    Compiling shlex v1.3.0
rustc>     Checking core v0.0.0 (/build/rustc-1.84.0-src/library/core)
rustc>    Compiling libc v0.2.162
rustc>    Compiling std v0.0.0 (/build/rustc-1.84.0-src/library/std)
rustc>  Documenting core v0.0.0 (/build/rustc-1.84.0-src/library/core)
rustc>    Compiling cc v1.1.22
rustc>    Compiling compiler_builtins v0.1.138
rustc>     Checking rustc-std-workspace-core v1.99.0 (/build/rustc-1.84.0-src/library/rustc-std-workspace-core)
rustc>     Checking alloc v0.0.0 (/build/rustc-1.84.0-src/library/alloc)
rustc>     Checking cfg-if v1.0.0
rustc>     Checking rustc-demangle v0.1.24
rustc>  Documenting alloc v0.0.0 (/build/rustc-1.84.0-src/library/alloc)
rustc>     Checking unwind v0.0.0 (/build/rustc-1.84.0-src/library/unwind)
rustc>     Checking rustc-std-workspace-alloc v1.99.0 (/build/rustc-1.84.0-src/library/rustc-std-workspace-alloc)
rustc>     Checking panic_abort v0.0.0 (/build/rustc-1.84.0-src/library/panic_abort)
rustc>     Checking hashbrown v0.15.0
rustc>     Checking std_detect v0.1.5 (/build/rustc-1.84.0-src/library/stdarch/crates/std_detect)
rustc>  Documenting std v0.0.0 (/build/rustc-1.84.0-src/library/std)
rustc> warning: dropping unsupported crate type `dylib` for target `wasm32v1-none`
rustc> error[E0433]: failed to resolve: use of undeclared crate or module `dlmalloc`
rustc>   --> std/src/sys/alloc/wasm.rs:24:43
rustc>    |
rustc> 24 | static mut DLMALLOC: dlmalloc::Dlmalloc = dlmalloc::Dlmalloc::new();
rustc>    |                                           ^^^^^^^^ use of undeclared crate or module `dlmalloc`
rustc> error[E0425]: cannot find function `fill_bytes` in module `sys`
rustc>   --> std/src/random.rs:67:14
rustc>    |
rustc> 67 |         sys::fill_bytes(bytes)
rustc>    |              ^^^^^^^^^^ not found in `sys`
rustc> error[E0433]: failed to resolve: could not resolve path `dlmalloc::Dlmalloc::new`
rustc>   --> std/src/sys/alloc/wasm.rs:24:43
rustc>    |
rustc> 24 | static mut DLMALLOC: dlmalloc::Dlmalloc = dlmalloc::Dlmalloc::new();
rustc>    |                                           ^^^^^^^^^^^^^^^^^^^^^^^ could not resolve path `dlmalloc::Dlmalloc::new`
rustc>    |
rustc>    = note: this error was originally ignored because you are running `rustdoc`
rustc>    = note: try running again with `rustc` or `cargo check` and you may get a more detailed error
rustc> error[E0433]: failed to resolve: use of undeclared crate or module `dlmalloc`
rustc>   --> std/src/sys/alloc/wasm.rs:24:22
rustc>    |
rustc> 24 | static mut DLMALLOC: dlmalloc::Dlmalloc = dlmalloc::Dlmalloc::new();
rustc>    |                      ^^^^^^^^ use of undeclared crate or module `dlmalloc`
rustc> For more information about this error, try `rustc --explain E0433`.
rustc> error: could not document `std`
rustc> warning: build failed, waiting for other jobs to finish...
rustc> error[E0425]: cannot find function `fill_bytes` in this scope
rustc>   --> std/src/sys/random/mod.rs:94:5
rustc>    |
rustc> 94 |     fill_bytes(&mut buf);
rustc>    |     ^^^^^^^^^^ not found in this scope
rustc> Some errors have detailed explanations: E0425, E0433.
rustc> For more information about an error, try `rustc --explain E0425`.
rustc> warning: `std` (lib) generated 1 warning
rustc> error: could not compile `std` (lib) due to 4 previous errors; 1 warning emitted
rustc> Command has failed. Rerun with -v to see more details.
rustc> Build completed unsuccessfully in 0:01:36
rustc> make: *** [Makefile:20: all] Error 1
```

--- a/pkgs/development/compilers/rust/rustc.nix
+++ b/pkgs/development/compilers/rust/rustc.nix
@@ -161,6 +161,7 @@ stdenv.mkDerivation (finalAttrs: {
           # Other targets that don't need any extra dependencies to build.
           ++ optionals (!fastCross) [
             "wasm32-unknown-unknown"
+            "wasm32v1-none"
             "bpfel-unknown-none"
             "bpfeb-unknown-none"
           ]

